src/pages/workspaces/workspace/ShareWorkspaceModal.js

```diff
@@ -79,41 +76,52 @@ export default ajaxCaller(class ShareWorkspaceModal extends Component {
       okButton: buttonPrimary({ onClick: () => this.save() }, ['Save']),
       onDismiss
     }, [
-      div({ style: styles.searchArea }, [
-        h(AutocompleteSearch, {
-          autoFocus: true,
-          placeholder: 'Add people or groups',
-          value: searchValue,
-          onChange: v => this.setState({ searchValue: v }),
-          renderSuggestion: suggestion => div({ style: styles.suggestionContainer }, [
-            div({ style: styles.suggestion }, [
-              !canAdd(suggestion) && h(TooltipTrigger, {
-                content: 'Not a valid email address'
-              }, [
-                icon('warning-standard', { style: { color: colors.red[0], marginRight: '0.5rem' } })
-              ]),
-              suggestion
-            ])
-          ]),
-          onSuggestionSelected: selection => {
-            if (canAdd(selection)) {
-              this.addAcl(selection)
-            }
-          },
-          onKeyDown: e => {
-            // 13 = Enter, 27 = Escape
-            if (e.which === 27 && !!searchValue) {
-              this.setState({ searchValue: '' })
-              e.stopPropagation()
-            } else if (e.which === 13 && !searchValueInvalid) {
-              this.addAcl(searchValue)
-            }
-          },
-          suggestions: _.difference(suggestions, _.map('email', acl)),
-          style: { fontSize: 16 },
-          theme: { suggestion: { padding: 0 } }
-        })
-      ]),
+      Forms.formLabel('User email'),
+      h(AutocompleteSearch, {
+        autoFocus: true,
+        placeholder: 'Add people or groups',
+        value: searchValue,
+        onChange: v => this.setState({ searchValue: v }),
+        renderSuggestion: suggestion => div({ style: styles.suggestionContainer }, [
+          div({ style: { flex: 1 } }, [
+            !canAdd(suggestion) && h(TooltipTrigger, {
+              content: 'Not a valid email address'
+            }, [
+              icon('warning-standard', {
+                style: { color: colors.red[0], marginRight: '0.5rem' }
+              })
+            ]),
+            suggestion
+          ])
+        ]),
+        onSuggestionSelected: selection => {
+          this.setState({ searchValue: selection })
+        },
+        onKeyDown: e => {
+          // 27 = Escape
+          if (e.which === 27 && !!searchValue) {
+            this.setState({ searchValue: '' })
+            e.stopPropagation()
+          }
+        },
+        suggestions: _.difference(suggestions, _.map('email', acl)),
+        style: { fontSize: 16 },
+        renderInputComponent: textInput,
```

Comment:
My impression was that we wanted to be consistent with how we add users to billing projects and groups--both of which use `AutocompleteSearch` (and also have the same console error when clicking a suggestion). If we wanted to change it to `AutocompleteTextInput` in both places, that could be done too?

Reply:
The overall UX is good, but the console error is obviously a problem. For this feature, we should either fix the error directly, or use a different component that doesn't exhibit the error. (There are slight differences in UX between the two components, so that would probably end up driving the decision).

Additionally, we should file a ticket to fix the error on the other page.